src/overseer/core.clj

```diff
@@ -111,8 +111,8 @@
     to be idempotent. Returns :ok on success")
 
   (transact-graph [this graph]
-    "Given a Graph, atomically transact all of its jobs/dependencies into the store.
-    Side-effecting only; return value is undefined (raises if transaction fails)")
+    "Given a Graph, atomically transact all of its jobs/dependencies into the store and
+    return it. Idempotent on job-ids in graph, will not double-insert.")
```

Comment:
Noted below, but especially since this is now part of the generic interface-level contract I'd expect a generic interface-level test.